### PR TITLE
Skip over unsupported entity types.

### PR DESCRIPTION
**Pull request by julien-me**
This PR makes it possible to read entity data packets containing entity data of any type by skipping over types that are not yet implemented.

### CODE REVIEW

File: src/domain/entities/GizmoEntityItem.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    bytesRead: number;
+    properties: GizmoEntitySubclassProperties;
+};
+
+
+class GizmoEntityItem {
+    // C++  class GizmoEntityItem : public EntityItem
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```
Comment (ctrlaltdavid) [inline src/domain/entities/GizmoEntityItem.ts:34]:
```suggestion
```
These appear to be not needed. Here and elsewhere.

File: src/domain/entities/PolyVoxEntityItem.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            // WEBRTC TODO: Read voxelVolumeSize property.
+            dataPosition += 12;
+        }
+
+        if (propertyFlags.getHasProperty(EntityPropertyFlags.PROP_VOXEL_DATA)) {
+            const length = data.getUint16(dataPosition, UDT.LITTLE_ENDIAN);
+            dataPosition += 2;
+
+            if (length > 0) {
+                for (let j = 0; j < length; j++) {
```
Comment (ctrlaltdavid) [inline src/domain/entities/PolyVoxEntityItem.ts:52]:
```suggestion
                for (let i = 0; i < length; i++) {
```
Outermost for loop should use index variable "i". Here and elsewhere.

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by ctrlaltdavid** [comment]
- The example app generates errors when connected to a domain with Box or Sphere entities in it.
- In `onEntityData(data)`, what does the `if (entityIDsList.some((id)` branch do, why is it needed? Code comment needed.
- In `onEntityData(data)`, let's use constants instead of magic numbers. See `X_INDEX` in the avatar mixer code, for example.

File: src/domain/entities/GizmoEntityItem.ts (hunk 1 above)
Comment (julien-me) [reply src/domain/entities/GizmoEntityItem.ts:34]: Correct, in `GizmoEntityItem.ts` and `GridEntityItem.ts` it is not needed.

But in the following files `data` is not used:
`BloomPropertyGroup.ts`, `HazePropertyGroup.ts`, `LightEntityItem.ts`, `PulsePropertyGroup.ts` and `RingGizmoPropertyGroup.ts`.

**Comment by julien-me** [comment]
> * The example app generates errors when connected to a domain with Box or Sphere entities in it.
What is the error and/or how to reproduce it? I don't see any error with Box, Sphere or other entities.

>     * In `onEntityData(data)`, what does the `if (entityIDsList.some((id)` branch do, why is it needed? Code comment needed.
We maintain a list of entity data already received.  The server sends entity data only for the entities that have been edited. When receiving new data the example app checks whether an entity with the same ID already exists in the list and updates it, or creates a new row in the table if it's a new ID.

**Comment by ctrlaltdavid** [comment]
Here are the Interface errors I'm getting:
image

**Comment by julien-me** [comment]
> Here are the Interface errors I'm getting: image

For some reason sometimes an entity's first packet is sent with minimal properties (ID, name, createdTime, etc). I modified the example app to check that a property exists before creating/updating a row in the table.

**Comment by ctrlaltdavid** [comment]
It looks like `EntityItem::appendEntityData()` will send partial data and flag which properties it managed to fit into the packet.

I see in the JSDoc that most property values may be `undefined` - presumably because of this?

It could be a good idea to add a comment to the top of each `EntityProperties` table along the lines of "a property value may be undefined if it couldn't fit in the data packet sent by the server"?

**Comment by julien-me** [comment]
> It looks like `EntityItem::appendEntityData()` will send partial data and flag which properties it managed to fit into the packet.
> 
> I see in the JSDoc that most property values may be `undefined` - presumably because of this?
> 
> It could be a good idea to add a comment to the top of each `EntityProperties` table along the lines of "a property value may be undefined if it couldn't fit in the data packet sent by the server"?

I've added it to the JSDoc.

**Review by ctrlaltdavid — APPROVED** [review]

**Pull request merged**